Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -9,160 +9,37 @@
-				{
-					HkKickReason(ARG_CLIENTID(checkDockedClientID), L"Forced undocking.");
-
-					// If carrier died, we don't need this message.
-					if (!mobiledockClients[checkDockedClientID].carrierDied)
-						PrintUserCmdText(checkCarrierClientID, L"Jettisoning ship...");
-				}
-			}
-
-			it = jettisonList.erase(it);
-		}
-		else
-		{
-			it++;
-		}
-	}
-}
-
-// Easy creation of delayed action.
-void DelayedJettison(int delayTimeSecond, wstring carrierCharname, wstring dockedCharname)
-{
-	ActionJettison action;
-	action.timeLeft = delayTimeSecond;
-	action.carrierCharname = carrierCharname;
-	action.dockedCharname = dockedCharname;
-	jettisonList.push_back(action);
-}
-
-void JettisonShip(uint carrierClientID, uint dockedClientID)
-{
-	const wchar_t* carrierCharname = (const wchar_t*)Players.GetActiveCharacterName(carrierClientID);
-	const wchar_t* dockedCharname = (const wchar_t*)Players.GetActiveCharacterName(dockedClientID);
-
-	if (dockedClientID != -1)
-	{
-		if (!mobiledockClients[dockedClientID].carrierDied)
-		{
-			PrintUserCmdText(carrierClientID, L"Ship warned. If it doesn't undock in %i seconds, it will be kicked by force.", jettisonKickTime);
-			PrintUserCmdText(dockedClientID, L"Carrier wants to jettison your ship. Undock willingly or you will be kicked after %i seconds.", jettisonKickTime);
-		}
-		else
-		{
-			PrintUserCmdText(dockedClientID, L"ALERT! Carrier ship is being destroyed. Launch ship or you will die in explosion.");
-		}
-		pub::Audio::PlaySoundEffect(dockedClientID, CreateID("rtc_klaxon_loop"));
-
-		// Create delayed action.
-		DelayedJettison(jettisonKickTime, carrierCharname, dockedCharname);
-	}
-}
-
-// Returns count of installed docking modules on ship of specific client.
-uint GetInstalledModules(uint iClientID)
-{
-	uint modules = 0;
-
-	// Check to see if the vessel undocking currently has a docking module equipped
-	for (list<EquipDesc>::iterator item = Players[iClientID].equipDescList.equip.begin(); item != Players[iClientID].equipDescList.equip.end(); item++)
-	{
-		if (find(dockingModuleEquipmentIds.begin(), dockingModuleEquipmentIds.end(), item->iArchID) != dockingModuleEquipmentIds.end())
-		{
-			if (item->bMounted)
-			{
-				modules++;
-			}
-		}
-	}
-
-	return modules;
-}
-
-void UpdateCarrierLocationInformation(uint dockedClientId, uint carrierShip)
-{
-
-	// Prepare references to the docked ship's copy of the carrierShip's position and rotation for manipulation
-	Vector& carrierPos = mobiledockClients[dockedClientId].carrierPos;
-	Matrix& carrierRot = mobiledockClients[dockedClientId].carrierRot;
+unordered_map<wstring, CARRIERINFO> nameToCarrierInfoMap;
+unordered_map<wstring, DOCKEDCRAFTINFO> nameToDockedInfoMap;
+unordered_map<uint, CARRIERINFO*> idToCarrierInfoMap;
+unordered_map<uint, DOCKEDCRAFTINFO*> idToDockedInfoMap;
 
+uint saveFrequency = 7200; // once every 2 hours, possibly a heavy operation so doesn't have to happen often

```

Comment:
Save doesn't denote what is actually being saved. 2 hours is also quite arbitrary. Saving to files can be delegated to threads without much work, I would suggest doing that.